### PR TITLE
Remove load test data

### PR DESCRIPTION
**Pull request by TobyRet**
_Creation of a rake task that removes data generated by load tests._

**NB: to be run on staging only.** 

- Guard clause has been added. 
- An env var ('REMOVE_LOAD_TEST_DATA') will need to be set for staging environment

Corresponding PR for cron job - https://github.com/ministryofjustice/pvb2-deploy/pull/172

### CODE REVIEW

File: spec/services/load_test_data_remover_spec.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        expect { described_class.run }.
+          not_to change { [Visit.count, Prisoner.count] }
+      end
+    end
+
+    context "with 'remove_load_test_data' flag switched on" do
+      context "when visitor is not 'Load Test'" do
+        it 'does not remove them' do
+          remove_load_test_data(:on)
+          # visitor.update!(first_name: first_name, last_name: last_name)
```
Comment (StupidCodeFactory) [inline spec/services/load_test_data_remover_spec.rb]: should this be removed?

File: spec/services/load_test_data_remover_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  context 'when visit request not generated by load tests' do
+    let(:first_name) { 'Load' }
+    let(:last_name)  { 'Spencer' }
+    let(:visitor)    { vst.visitors.last }
+
+    context 'with "remove_load_test_data" flag switched off' do
+      it 'does not remove them' do
+        remove_load_test_data(:off)
+        # visitor.update!(first_name: first_name, last_name: last_name)
```
Comment (StupidCodeFactory) [inline spec/services/load_test_data_remover_spec.rb]: should this be removed?

File: spec/services/load_test_data_remover_spec.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  let(:prisoner_dob) { '1979-04-23' }
+  let(:visitor) { vst.visitors.last }
+
+  context 'when visit requests are generated by load tests' do
+    let(:first_name) { 'Load' }
+    let(:last_name)  { 'Test' }
+
+    context 'with "remove_load_test_data" flag switched on' do
+      it 'removes them' do
+        remove_load_test_data(:on)
```
Comment (StupidCodeFactory) [inline spec/services/load_test_data_remover_spec.rb]: We've already something similar you can use to stub the value of te configuration

https://github.com/ministryofjustice/prison-visits-2/blob/master/spec/support/helpers/configuration_helpers.rb

**Comment by TobyRet** [comment]
@StupidCodeFactory - thanks for review - have made the changes ...

File: spec/services/load_test_data_remover_spec.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          expect { described_class.run }.
+            not_to change { [Visit.count, Prisoner.count] }
+        end
+      end
+    end
+  end
+
+  def visit_for(first_name, last_name)
+    Visit.joins(:visitors).
+      where("visitors.first_name = '#{first_name}'", "visitors.last_name = '#{last_name}'")
```
Comment (StupidCodeFactory) [inline spec/services/load_test_data_remover_spec.rb]: Even if this works, it's a bad habbit to take as aI don't think SQL sanitisation will happen.
Prefer the following:
```ruby
Visit.joins(:visitors).
     where(visitors: { first_name: first_name, last_name: last_name })
```

Comment (StupidCodeFactory) [reply spec/services/load_test_data_remover_spec.rb]: @TobyRet sorry I missed this the first time around

**Pull request merged**